rskj-core/src/test/java/co/rsk/peg/BridgeSupportAddSignatureTest.java

```diff
@@ -0,0 +1,727 @@
+    }
+
+    @Test
+    public void addSignatureWithMoreSignaturesThanExpected() throws Exception {
+        List<BtcECKey> keys = Collections.singletonList(BridgeRegTestConstants.REGTEST_FEDERATION_PRIVATE_KEYS.get(0));
+        addSignatureFromValidFederator(keys, 2, true, false, "InvalidParameters");
+    }
+
+    @Test
+    public void addSignatureNonCanonicalSignature() throws Exception {
+        List<BtcECKey> keys = Collections.singletonList(BridgeRegTestConstants.REGTEST_FEDERATION_PRIVATE_KEYS.get(0));
+        addSignatureFromValidFederator(keys, 1, false, false, "InvalidParameters");
+    }
+
+    private void test_addSignature_EventEmitted(boolean rskip146Active, boolean rskip326Active, byte[] signature, int wantedNumberOfInvocations) throws Exception {
+        // Setup
+        Federation federation = bridgeConstantsRegtest.getGenesisFederation();
+        Repository track = createRepository().startTracking();
+        BridgeStorageProvider provider = new BridgeStorageProvider(track, PrecompiledContracts.BRIDGE_ADDR, bridgeConstantsRegtest, activationsBeforeForks);
+
+        // Build prev btc tx
+        BtcTransaction prevTx = new BtcTransaction(btcRegTestParams);
+        TransactionOutput prevOut = new TransactionOutput(btcRegTestParams, prevTx, Coin.FIFTY_COINS, federation.getAddress());
+        prevTx.addOutput(prevOut);
+
+        // Build btc tx to be signed
+        BtcTransaction btcTx = new BtcTransaction(btcRegTestParams);
+        btcTx.addInput(prevOut).setScriptSig(createBaseInputScriptThatSpendsFromTheFederation(federation));
+        TransactionOutput output = new TransactionOutput(btcRegTestParams, btcTx, Coin.COIN, new BtcECKey().toAddress(btcRegTestParams));
+        btcTx.addOutput(output);
+
+        // Save btc tx to be signed
+        final Keccak256 rskTxHash = createHash3(1);
+        provider.getRskTxsWaitingForSignatures().put(rskTxHash, btcTx);
+        provider.save();
+        track.commit();
+
+        // Setup BridgeSupport
+        BridgeEventLogger eventLogger = mock(BridgeEventLogger.class);
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+
+        when(activations.isActive(ConsensusRule.RSKIP146)).thenReturn(rskip146Active);
+        when(activations.isActive(ConsensusRule.RSKIP326)).thenReturn(rskip326Active);
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+                .withBridgeConstants(bridgeConstantsRegtest)
+                .withProvider(provider)
+                .withRepository(track)
+                .withEventLogger(eventLogger)
+                .withBtcLockSenderProvider(new BtcLockSenderProvider())
+                .withExecutionBlock(mock(Block.class))
+                .withBtcBlockStoreFactory(null)
+                .withActivations(activations)
+                .build();
+
+        BtcECKey privateKeyToSignWith = BridgeRegTestConstants.REGTEST_FEDERATION_PRIVATE_KEYS.get(0);
```

Comment:
```suggestion
        int indexOfKeyToSignWith = 0;
        BtcECKey privateKeyToSignWith = BridgeRegTestConstants.REGTEST_FEDERATION_PRIVATE_KEYS.get(indexOfKeyToSignWith);
```
Then we can use this same index value to get the public key